src/lib/Libecl/ecl_verify_values.c

```diff
@@ -1105,13 +1105,15 @@ int
 verify_value_zero_or_positive(int batch_request, int parent_object, int cmd,
 	struct attropl *pattr, char **err_msg)
 {
-	long l;
+	long lval;
+	char *end = NULL;
 
 	if ((pattr->value == NULL) || (pattr->value[0] == '\0'))
 		return PBSE_BADATVAL;
 
-	l = atol(pattr->value);
-	if (l < 0)
+	errno = 0;
+	lval = strtol(pattr->value, &end, 10);
+	if ((errno != 0) || (lval < 0))
```

Comment:
Mike gave a comment about setting the errno. I was also of the same opinion that we should not be setting any errno ourselves but then I saw strtol() man page and the example in the man page explicitly sets the errno. Besides, the caller isn't multithreaded so it shouldn't matter if I set it here.
About ERANGE, isn't it better to check for all errors instead of checking only for one specific error?
I'll try and check what happens in case of a hook. Life would have been much easier if hooks infrastructure was also using the same verification functions. In any case, since run_count is supposed to be zero or positive number there must be some check in hooks infra which would be doing this.

Reply:
I checked, in case of hooks we use "PyArg_Parse" to convert the input to a long and that function checks for overflow. We then also check that the value must be positive. So I think we are covered from hooks side